### PR TITLE
MON-3644: Ease the tracking of monitoring components versions.

### PR DESCRIPTION
**Pull request by machine424**
…ate_versions.go and provide components versions for multiple OCP versions

* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: hack/tools/generate_versions.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
+package main
```
Comment (machine424) [inline hack/tools/generate_versions.go:1]: It'd be great if we gradually migrate the other `.sh` scripts into Go for more readability and sanity :)
I'll create a ticket

Tell me if comments are needed for thei one.

**Comment by machine424** [comment]
/retitle MON-3644: Ease the tracking of monitoring components versions.

File: hack/tools/generate_versions.go (hunk 1 above)
Comment (simonpasquier) [reply hack/tools/generate_versions.go:1]: I'd rather keep hack/tools for tracking/installing external Go binaries that are needed by the build process. Can we move the program to a dedicated subdirectory?

File: hack/tools/generate_versions.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+	"github.com/olekukonko/tablewriter"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	mainBranch        = "master"
+	metricsServerRepo = "openshift/kubernetes-metrics-server"
+	versionFile       = "../../jsonnet/versions.yaml"
+	// This representation works for now.
+	minOCPVersion       = float32(4.10)
```
Comment (simonpasquier) [inline hack/tools/generate_versions.go:22]: In prometheus-operator, we use `github.com/blang/semver/v4` which is ok.

File: Documentation/deps-versions.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+|  OCP Version |                                       alertmanager                                       |                                  kubeRbacProxy                                   |                                  kubeStateMetrics                                   |                                                kubernetesMetricsServer                                                 |                                 monitoringPlugin                                  |                                 nodeExporter                                  |                                  promLabelProxy                                  |                                 prometheus                                  |                                    prometheusAdapter                                    |                                  prometheusOperator                                  |                                 thanos                                  |
```
Comment (simonpasquier) [inline Documentation/deps-versions.md:1]: I wonder if it wouldn't be more readable to have dedicated sections per minor?

## 4.16

| Component | Version |
|--------|--------|
| Alertmanager | 0.26.0 |
| kube-rbac-proxy | 0.15.0 |
| kube-state-metrics | 2.10.1 | 
| ... | ... | 

## 4.15

...

File: Documentation/deps-versions.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+| release-4.10 | [0.23.0](https://github.com/openshift/prometheus-alertmanager/blob/release-4.10/VERSION) | [0.11.0](https://github.com/openshift/kube-rbac-proxy/blob/release-4.10/VERSION) | [2.3.0](https://github.com/openshift/kube-state-metrics/blob/release-4.10/VERSION)  | N/A                                                                                                                    | N/A                                                                               | [1.3.1](https://github.com/openshift/node_exporter/blob/release-4.10/VERSION) | [0.4.0](https://github.com/openshift/prom-label-proxy/blob/release-4.10/VERSION) | [2.32.1](https://github.com/openshift/prometheus/blob/release-4.10/VERSION) | [0.9.1](https://github.com/openshift/k8s-prometheus-adapter/blob/release-4.10/VERSION)  | [0.53.1](https://github.com/openshift/prometheus-operator/blob/release-4.10/VERSION) | [0.23.2](https://github.com/openshift/thanos/blob/release-4.10/VERSION) |
```
Comment (simonpasquier) [inline Documentation/deps-versions.md:3]: (nit) The link could be to the branch's root directory (e.g. https://github.com/openshift/prometheus-alertmanager/blob/release-4.10).

Comment (machine424) [reply Documentation/deps-versions.md:3]: I point it to the file containing the version to ease the retrieval of the commit that upgraded the version.

File: hack/tools/generate_versions.go (hunk 2 above)
Comment (machine424) [reply hack/tools/generate_versions.go:22]: `4.10` isn't really a semver... But I pushed a version using it by ignoring the `patch`

File: Documentation/deps-versions.md (hunk 3 above)
Comment (machine424) [reply Documentation/deps-versions.md:1]: You'd need to scroll down a lot for older versions and the unique table easier to generate...

File: hack/tools/generate_versions.go (hunk 1 above)
Comment (machine424) [reply hack/tools/generate_versions.go:1]: You're right, I thought I put them next to the sh ones...
I put them in a go/ (not too creative :)) dir now.

File: jsonnet/versions.yaml [hunk 5]
```diff
@@ -1,28 +1,28 @@
 ---
-# This file is meant to be managed by hack/generate-versions.sh script
+# This file is meant to be managed by hack/tools/generate_versions.go script
```
Comment (jan--f) [inline jsonnet/versions.yaml:2]:
```suggestion
# This file is meant to be managed by hack/go/generate_versions.go script
```

File: Documentation/deps-versions.md (hunk 3 above)
Comment (jan--f) [reply Documentation/deps-versions.md:1]: If we go with a single table (no strong opinion on that) can we have the newest version on top?

File: Documentation/deps-versions.md (hunk 4 above)
Comment (jan--f) [reply Documentation/deps-versions.md:3]: +1 for linking the repo root. For retrieval you could just hardcode the `VERSION` path component, no?

File: Documentation/deps-versions.md (hunk 3 above)
Comment (machine424) [reply Documentation/deps-versions.md]: newest versions are on top now.

File: Documentation/deps-versions.md (hunk 4 above)
Comment (machine424) [reply Documentation/deps-versions.md]: done.

File: jsonnet/versions.yaml (hunk 5 above)
Comment (machine424) [reply jsonnet/versions.yaml]: thanks

**Comment by jan--f** [comment]
Thanks! This
/lgtm

**Pull request merged**